src/GraphManager/GraphEdit/ButtonTests.test.tsx

```diff
@@ -0,0 +1,178 @@
+import { makeMockController } from "./GraphEdit.testingutil";
+import {openCreateNodePopUpAtPagePosition, openCreateLinkPopUp } from "./GraphEdit";
+
+jest.mock("./GraphEdit", () => ({
+    openCreateNodePopUpAtPagePosition: jest.fn(),
+    openCreateLinkPopUp: jest.fn(),
+  }));
+
+describe("UserSettings", () => {
+    it("should toggle isOpen state when toggleDrawer is called", () => {
+        // Arrange: Mock initial state
+        let isOpen = false;
+        const toggleDrawer = () => {
+          isOpen = !isOpen; // Simulate toggling the drawer state
+        };
+        // Act: Simulate what happens when the button is clicked
+        toggleDrawer();
+      
+        // Assert: Check if the state toggled
+        expect(isOpen).toBe(true); 
+        // Act again: Toggle back to simulate closing the drawer
+        toggleDrawer();
+      
+        // Assert: Check if the state toggled back
+        expect(isOpen).toBe(false);
+      });
+      it("should call ctrl.mode.setUse3D when toggling 3D display", () => {
```

Comment:
this test also tests nothing.. should probably be removed
generally we don't want to test the react implementation, but our own code!